tests/microbenchmarks/bench_moe/test_v4_trace_discrete.py

```
@@ -0,0 +1,341 @@
+
+
+def _split_template_args(s: str) -> list[str]:
+    """Split a template argument list on commas that are not nested."""
+    out, depth, cur = [], 0, []
+    for ch in s:
+        if ch in "<([":
+            depth += 1
+        elif ch in ">)]":
+            depth -= 1
+        if ch == "," and depth == 0:
+            out.append("".join(cur).strip()); cur = []
```

Ruff (0.16.1)

[error] Multiple statements on one line (semicolon)

(E702)